File: materials/ZeroJudge/r489/r489_stupid_approach.py
def rotate90(a):
    r = len(a)
    c = len(a[0])

    result = [[0] * r for _ in range(c)]

    for i in range(r):
        for j in range(c):
            result[j][r - 1 - i] = a[i][j]

    return result


def compare(a, b):
    # 尺寸不一樣，不能直接比較
    if len(a) != len(b):
        return -1

    if len(a[0]) != len(b[0]):
        return -1

    same = 0

    for i in range(len(a)):
        for j in range(len(a[0])):
            if a[i][j] == b[i][j]:
                same += 1

    return same


r, c = map(int, input().split())

lst = []
pic = []

for i in range(r):
    lst.append(list(map(int, input().split())))

for i in range(r):
    pic.append(list(map(int, input().split())))


# 儲存四種旋轉
lst_rotate = [lst]
pic_rotate = [pic]

# 每次把上一張再轉 90°
for i in range(1, 4):
    lst_rotate.append(rotate90(lst_rotate[i - 1]))
    pic_rotate.append(rotate90(pic_rotate[i - 1]))


best = 0

# 4 × 4 = 16 種組合全部試
for a in range(4):
    for b in range(4):
        same = compare(lst_rotate[a], pic_rotate[b])

        if same != -1:
            best = max(best, same)


print(str(best * 100 // (r * c)) + "%")

File: materials/ZeroJudge/r489/r489_stupid_approach.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

vector<vector<ll>> rotate90(vector<vector<ll>> a)
{
    ll r = a.size();
    ll c = a[0].size();

    vector<vector<ll>> result(c, vector<ll>(r));

    for (ll i = 0; i < r; i++)
    {
        for (ll j = 0; j < c; j++)
        {
            result[j][r - 1 - i] = a[i][j];
        }
    }

    return result;
}

ll compare(vector<vector<ll>>& a, vector<vector<ll>>& b)
{
    // 尺寸不一樣，不能直接疊在一起比較
    if (a.size() != b.size())
        return -1;

    if (a[0].size() != b[0].size())
        return -1;

    ll same = 0;

    for (ll i = 0; i < a.size(); i++)
    {
        for (ll j = 0; j < a[0].size(); j++)
        {
            if (a[i][j] == b[i][j])
            {
                same++;
            }
        }
    }

    return same;
}

int main()
{
    ll r, c;
    cin >> r >> c;

    vector<vector<ll>> lst(r, vector<ll>(c));
    vector<vector<ll>> pic(r, vector<ll>(c));

    for (ll i = 0; i < r; i++)
    {
        for (ll j = 0; j < c; j++)
        {
            cin >> lst[i][j];
        }
    }

    for (ll i = 0; i < r; i++)
    {
        for (ll j = 0; j < c; j++)
        {
            cin >> pic[i][j];
        }
    }

    // 儲存四種旋轉
    vector<vector<vector<ll>>> lst_rotate(4);
    vector<vector<vector<ll>>> pic_rotate(4);

    lst_rotate[0] = lst;
    pic_rotate[0] = pic;

    // 每次都把上一張再轉 90°
    for (ll i = 1; i < 4; i++)
    {
        lst_rotate[i] = rotate90(lst_rotate[i - 1]);
        pic_rotate[i] = rotate90(pic_rotate[i - 1]);
    }

    ll best = 0;

    // 4 × 4 = 16 種組合全部試
    for (ll a = 0; a < 4; a++)
    {
        for (ll b = 0; b < 4; b++)
        {
            ll same = compare(lst_rotate[a], pic_rotate[b]);

            if (same != -1)
            {
                best = max(best, same);
            }
        }
    }

    cout << best * 100 / (r * c) << "%" << endl;
}